packages/client/src/components/Outcome/NewCheckerModal.tsx

```diff
@@ -42,115 +32,59 @@ const NewCheckerModal: FunctionComponent = () => {
       action: actions.OPEN_MODAL,
       name: eventNames.OPEN_MODAL_DO_ANOTHER_CHECK,
     });
-
-    setError(false);
   };
 
   const handleConfirmButton = () => {
-    const hasError = saveAddress === null;
-
-    setError(hasError);
+    setFinished(!!checkerSlug);
 
-    if (hasError) {
-      matomoTrackEvent({
-        action: actions.START_ANOTHER_CHECK,
-        name: `${eventNames.DO_ANOTHER_CHECK} - ${eventNames.NO_CHOICE_HAS_BEEN_MADE}`,
-      });
-    } else {
-      setFinished(!!checkerSlug);
+    matomoTrackEvent({
+      action: actions.START_ANOTHER_CHECK,
+      name: eventNames.DO_ANOTHER_CHECK,
+    });
 
-      const doSaveAddress = saveAddress === true;
+    // Set the new topic data with the correct `type` and potentially the `address`
+    const newTopicData = {
+      ...defaultTopicSession,
+      address: topicData.address,
+      // Update timesCheckerLoaded + 1 or reset to zero if going to a new checker
+      timesCheckerLoaded:
+        checkerSlug === slug ? topicData.timesCheckerLoaded + 1 : 0,
+      type: checkerSlug,
+    };
+
+    // Clear checker
+    setChecker(undefined);
```

Comment:
For performance it might be better to place `setChecker(undefined)` underneath each `history.push`. Maybe you can test?